tutorial.md

```diff
@@ -0,0 +1,165 @@
+curl -L -o - faas.gz https://github.com/boson-project/faas/releases/download/v0.6.0/faas_linux_amd64.gz | gunzip > faas && chmod 755 faas
+sudo mv faas /usr/local/bin
+```
+## Configuring a Container Repository
+The unit of deployment in Boson Functions is an [OCI](https://opencontainers.org/)
+container image, typically referred to as a Docker container image.
+
+In order for the `faas` CLI to manage these containers, you'll need to be
+logged in to a container registry.
+
+
+```bash
+docker login -u <username> -p <password> <repository.url>
+```
+
+> Note: many of the CLI commands take a `--repository` argument. Set the `FAAS_REPOSITORY` environment variable in order to omit this parameter when using the CLI.
+
+```bash
+export FAAS_REPOSITORY=docker.io/<repo.name>
+```
+
+## Creating a Project
+With your Knative enabled cluster up and running, you can now create a new
+Function Project. Let's start by creating a project directory. Function names
+in `faas` correspond to URLs at the moment, and there are some finicky cases
+at the moment. To ensure that everything works as it should, create a project
+directory consisting of three URL parts. Here is a good one.
+
+```bash
+mkdir fn.example.io
+cd fn.example.io
+```
+
+Now, with one command we will create the project files, build a container, and
+deploy the function as a Knative service.
+
+
+```bash
+faas create -l node -y
+```
+
+This will create a Node.js Function project in the current directory accepting
+all of the defaults inferred from your environment, for example`$FAAS_REPOSITORY`.
+When the command has completed, you can see the deployed function.
+
+```bash
+$ kn service list
+NAME            URL                                          LATEST                  AGE   CONDITIONS   READY   REASON
+fn-example-io   http://fn-example-io.faas.127.0.0.1.nip.io   fn-example-io-ngswh-1   24s   3 OK / 3     True
+```
+
+Clicking on the URL will take you to the running function in your cluster.
+
+## Local Development
+The `faas create` command also results in a docker container that can be run locally
+with container ports mapped to localhost.
+
+```bash
+faas run
+```
+
+For day to day development of the function, you can also run it locally outside of a
+container. For this project, using Node.js, you have the following commands available.
+
+```bash
+npm install # Installs all dependencies
+npm test # Runs unit and integration test suites
+npm run local # Execute the function on the local host
+```
+
+## Deploying to a Cluster - Step by Step
+With `faas create` you have already deployed to a cluster! But there was
+a lot of magic in that one command. Let's break it down step by step using
+the `faas CLI` to take each step in turn.
+
+First, let's delete the project we just created.
+
+```bash
+faas delete
+```
+
+You might see a message such as this.
+
+```bash
+Error: remover failed to delete the service: timeout: service 'fn-example-io' not ready after 30 seconds.
```

Comment:
I actually got this error:
```
Error: remover failed to delete the service: services.serving.knative.dev "fn-example-io" not found.
```

Reply:
Hmm - had you successfully created the project previously?

Reply:
i believe so.  I even clicked on the service link to open the web page, which only showed a blank "query" json object,  which i can't paste here because i closed the page.   

These are some of the commands that i did for the delete:

```
~/develop/boson-getting-started/fn.example.io » faas delete                                                                   lucasholmquist@lincolnhawk2
Function to remove (fn.example.io): yes
Error: : Invalid value: "yes": should be a domain with at least two segments separated by dots
----------------------------------------------------------------------------------------------------------------------------------------------------------
~/develop/boson-getting-started/fn.example.io » faas delete                                                               1 ↵ lucasholmquist@lincolnhawk2
Function to remove (fn.example.io): 
Error: remover failed to delete the service: services.serving.knative.dev "fn-example-io" not found.
StdOut: 
----------------------------------------------------------------------------------------------------------------------------------------------------------
~/develop/boson-getting-started/fn.example.io » kn service list                                                           1 ↵ lucasholmquist@lincolnhawk2
NAME            URL                                             LATEST                  AGE     CONDITIONS   READY   REASON
fn-example-io   http://fn-example-io.default.127.0.0.1.nip.io   fn-example-io-qgnhk-1   6m29s   3 OK / 3     True 
```

Reply:
@lance  There is a bug in the code I am about to submit a PR.

Reply:
just saw that it was merged,  i'm assuming this will be 0.6.3 once it is released?

Reply:
Should be, yes...

Reply:
getting the same thing with 0.6.2